src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -533,7 +533,12 @@ private sealed class Request
 
             public Request()
             {
-                _taskCompletionSource = new TaskCompletionSource<T>();
+                // We want to always run continuations asynchronously. Running them synchronously could result in deadlocks:
+                // if we're looping through a bunch of Requests and completing them one by one, and the continuation for the
+                // first Request was then blocking waiting for a later Request, we would hang. It also could cause performance
+                // issues. If the first request then consumes a lot of CPU time, we're not letting other Requests complete that
+                // could use another CPU core at the same time.
+                _taskCompletionSource = new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
does the TCS need to be a mutable field?  Also, can we move this right to the field initializer?